ledger/narwhal/batch-certificate/src/bytes.rs

```diff
@@ -62,14 +62,13 @@ impl<N: Network> FromBytes for BatchCertificate<N> {
                     Self::MAX_SIGNATURES
                 )));
             }
+            // Read the signature bytes.
+            let mut signature_bytes = vec![0u8; num_signatures as usize * Signature::<N>::size_in_bytes()];
+            reader.read_exact(&mut signature_bytes)?;
             // Read the signatures.
-            let mut signatures = IndexSet::with_capacity(num_signatures as usize);
-            for _ in 0..num_signatures {
-                // Read the signature.
-                let signature = Signature::read_le(&mut reader)?;
-                // Insert the signature.
-                signatures.insert(signature);
-            }
+            let signatures = cfg_chunks!(signature_bytes, Signature::<N>::size_in_bytes())
+                .map(Signature::read_le)
+                .collect::<Result<IndexSet<_>, _>>()?;
```

Comment:
@ljedrz is this in your opinion clean? i.e. can we do better

Reply:
yes, I don't see any issues with this impl